config/mysqlpersister/mysqlpersister.go

```diff
@@ -0,0 +1,209 @@
+package mysqlpersister
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/golang/protobuf/proto"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/square/quotaservice/config"
+	"github.com/square/quotaservice/logging"
+	qsc "github.com/square/quotaservice/protos/config"
+)
+
+type MysqlPersister struct {
+	latestVersion int
+	db            *sqlx.DB
+	m             *sync.RWMutex
+
+	watcher         chan struct{}
+	shutdown        chan struct{}
+	fetcherShutdown chan struct{}
+
+	configs map[int]*qsc.ServiceConfig
+}
+
+type configRow struct {
+	Version int    `db:"Version"`
+	Config  string `db:"Config"`
+}
+
+func New(dbUser, dbPass, dbHost string, dbPort int, dbName string, pollingInterval time.Duration) (*MysqlPersister, error) {
+	db, err := sqlx.Open("mysql",
```

Comment:
What happens if the connection to the database is lost?

Reply:
If it's lost, it has a retry policy and then it should return.  I'm not sure if it will resume the connection after the database is back up

Reply:
Seems like the app is in charge of reconnecting according to [this](https://hackmongo.com/post/when-mysql-goes-away/)?